projects/core/src/routing/configurable-routes/configurable-routes.service.ts

```diff
@@ -49,75 +47,102 @@ export class ConfigurableRoutesService {
     return result;
   }
 
+  /**
+   * Sets the `path` or `matcher` of the `Route`, based on the Spartacus' configuration of the route.
+   */
   private configureRoute(route: Route): Route {
     const routeName = this.getRouteName(route);
     if (routeName) {
       const routeConfig = this.routingConfigService.getRouteConfig(routeName);
-      const paths = this.getConfiguredPaths(routeConfig, routeName, route);
+      this.validateRouteConfig(routeConfig, routeName, route);
+
+      const paths = (routeConfig && routeConfig.paths) || [];
       const isDisabled = routeConfig && routeConfig.disabled;
-      const matchers = routeConfig.matchers;
+      const matchers = routeConfig && routeConfig.matchers;
```

Comment:
`routeConfig?.matchers`

Reply:
we can actually drop the const and do the evaluation inline

Reply:
Thanks. Done.

Reply:
It's reused in 2 places so I'm happy to leave the const.

Reply:
I was hinting at:

```
 // const paths = (routeConfig?.paths) || [];
      // const isDisabled = routeConfig?.disabled;
      // const matchers = routeConfig?.matchers;

      if (routeConfig?.disabled) {
        delete route.path;
        return {
          ...route,
          matcher: this.urlMatcherService.getFalsy(),
        };
      } else if (routeConfig?.matchers) {
        delete route.path;
        return { ...route, matcher: this.resolveUrlMatchers(route, routeConfig.matchers) };
      } else if (routeConfig?.paths.length === 1) {
        delete route.matcher;
        return { ...route, path: routeConfig.paths[0] };
      } else {
        delete route.path;
        return {
          ...route,
          matcher: this.urlMatcherService.getFromPaths(routeConfig.paths),
        };
      }
```

leaving out those consts